services/brig/src/Brig/Run.hs

```diff
@@ -56,9 +60,13 @@ run o = do
     Async.async
       $ AWS.execute (e ^. awsEnv)
       $ AWS.listen throttleMillis q (runAppT e . SesNotification.onEvent)
+  sftDiscovery <- case e ^. sftEnv of
+    Nothing -> Async.async (pure ()) -- TODO: This looks fishy
+    Just sftEnv' -> Async.async $ Calling.startSFTServiceDiscovery sftEnv'
```

Comment:
Would it make sense to use a logic like in https://github.com/wireapp/wire-server/blob/develop/services/gundeck/src/Gundeck/ThreadBudget/Internal.hs#L243-L251 to restart that thread if it dies?

Reply:
Yeah, we need to implement something like that for all of the threads that we spawn in this function. I think this PR is running for too long, let's tackle this in https://github.com/zinfra/backend-issues/issues/1647.

Reply:
Sure, good idea.